monocle/config.py

```diff
@@ -25,71 +27,71 @@
                 },
             },
         },
-        "gerrit_repository": {
-            "$id": "http://monocle/gerrit_repository.schema.json",
-            "title": "Gerrit repository",
-            "description": "A gerrit repository description for the crawler",
+        "gitlab_provider": {
+            "title": "Gitlab provider",
             "type": "object",
-            "required": ["name", "updated_since", "base_url"],
+            "required": ["gitlab_token"],
             "properties": {
-                "name": {
+                "gitlab_token": {"type": "string"},
+            },
+        },
+        "gerrit_provider": {
+            "title": "Gerrit provider",
+            "type": "object",
+            "required": ["gerrit_url"],
+            "properties": {
+                "gerrit_url": {
                     "description": "The repository name or regexp",
                     "type": "string",
                 },
-                "updated_since": {
-                    "description": "The change updated since date (YYYY-mm-dd)",
-                    "type": "string",
-                    "format": "date",
-                },
-                "base_url": {
-                    "description": "Base url of the Gerrit instance",
-                    "type": "string",
-                },
-                "prefix": {
-                    "description": "Repository name prefix in case of name collision",
-                    "type": "string",
-                },
-                "insecure": {
+                "gerrit_url_insecure": {
                     "description": "Set to true to bypass the HTTP X509 certificate verification",
                     "type": "boolean",
                 },
-                "login": {
-                    "description": "Login to use to authenticate on Gerrit",
-                    "type": "string",
+                "gerrit_repositories": {
+                    "type": "array",
+                    "items": {"type": "string"},
                 },
-                "password": {
-                    "description": "Password to use to authenticate on Gerrit",
+                "gerrit_prefix": {
+                    "description": "Repository name prefix in case of name collision",
                     "type": "string",
                 },
             },
         },
+        "crawler": {
+            "title": "Monocle crawler",
+            "description": "A crawler definition",
+            "type": "object",
+            "required": ["name", "update_since", "provider"],
+            "properties": {
+                "name": {"type": "string"},
+                "update_since": {"type": "string", "format": "date"},
+                "provider": {
+                    "oneOf": [
+                        {"$ref": "#/definitions/gerrit_provider"},
+                        {"$ref": "#/definitions/github_provider"},
+                        {"$ref": "#/definitions/gitlab_provider"},
```

Comment:
Do we need the GitLab crawler here ?

Reply:
Turns out json schema is pretty flexible, and we might omit this list.